src/Mod/Arch/ArchRoof.py

```diff
@@ -42,241 +52,328 @@ def QT_TRANSLATE_NOOP(ctxt,txt):
+            try:
+                norm1.normalize()
+            except Part.OCCError:
+                return []
+
+        if not DraftVecUtils.isNull(norm2):
+            try:
+                norm2.normalize()
+            except Part.OCCError:
+                return []
+
+        if DraftVecUtils.isNull(norm1.cross(norm2)):
+            vec1 = pt2.sub(pt1)
+            vec2 = pt4.sub(pt3)
+            if DraftVecUtils.isNull(vec1) or DraftVecUtils.isNull(vec2):
+                return []  # One of the lines has zero-length
+            try:
+                vec1.normalize()
+                vec2.normalize()
+            except Part.OCCError:
+                return []
+            norm3 = vec1.cross(vec2)
+            denom = norm3.x + norm3.y + norm3.z
+            if not DraftVecUtils.isNull(norm3) and denom != 0:
+                k = ((pt3.z - pt1.z) * (vec2.x - vec2.y)
+                     + (pt3.y - pt1.y) * (vec2.z - vec2.x)
+                     + (pt3.x - pt1.x) * (vec2.y - vec2.z)) / denom
+                vec1.scale(k, k, k)
+                intp = pt1.add(vec1)
+
+                if infinite1 is False and not isPtOnEdge(intp, edge1):
+                    return []
+
+                if infinite2 is False and not isPtOnEdge(intp, edge2):
+                    return []
+
+                return [intp]
+            else:
+                return []  # Lines have same direction
+        else:
+            return []  # Lines aren't on same plane
 
+    pt1, pt2, pt3, pt4 = [edge1.Vertexes[0].Point,
+                          edge1.Vertexes[1].Point,
+                          edge2.Vertexes[0].Point,
+                          edge2.Vertexes[1].Point]
 
-def makeRoof(baseobj=None,facenr=0, angles=[45.,], run = [], idrel = [0,],thickness = [50.,], overhang=[100.,], name="Roof"):
+    return getLineIntersections(pt1, pt2, pt3, pt4, infinite1, infinite2)
 
-    '''makeRoof(baseobj,[facenr],[angle],[name]) : Makes a roof based on
-    a closed wire or an object. You can provide a list of angles, run,
-    idrel, thickness, overhang for each edges in the wire to define the
-    roof shape. The default for angle is 45 and the list is
-    automatically complete to match with number of edges in the wire.
-    If the base object is a solid the roof take the shape.'''
 
+def face_from_points(ptLst):
+    ptLst.append(ptLst[0])
+    # Use DraftVecUtils.removeDouble after append as it does not compare the first and last vector:
+    ptLst = DraftVecUtils.removeDoubles(ptLst)
+    ln = len(ptLst)
+    if ln < 4: # at least 4 points are required for 3 edges
+        return None
+    edgeLst = []
+    for i in range(ln - 1):
+        edge = Part.makeLine(ptLst[i], ptLst[i + 1])
+        edgeLst.append(edge)
+    wire = Part.Wire(edgeLst)
+    return Part.Face(wire)
+
+
+def makeRoof(baseobj=None,
+             facenr=0,
+             angles=[45.0], run=[250.0], idrel=[-1], thickness=[50.0], overhang=[100.0],
```

Comment:
Just realize that if you change the default values for `run` and `idrel`, you may be breaking the scripts of other people or the default behavior that other people were expecting, so it is important to understand the consequences.

Reply:
Yes, I do understand. That is the reason I have not renamed the properties or the arguments. Although they are strange: Why 'angles' (OK: plural because its a list), and 'run' (also a list, but singular).

Reply:
I think we can change the argument names in the future, but it has to be a planned deprecation. So if you add a comment about it, it can be addressed in the future.

Reply:
I don't think that makeRoof function is used anywhere else ATM. So this should be safe. But indeed if we change arg names we should announce it better somehow